src/lib/integration_test_lib/graphql_polling_log_engine.ml

```diff
@@ -0,0 +1,149 @@
+open Async
+open Core
+module Timeout = Timeout_lib.Core_time
+
+(** This implements Log_engine_intf for integration tests, by creating a simple system that polls a mina daemon's graphql endpoint for fetching logs*)
+
+module Make_GraphQL_polling_log_engine
```

Comment:
I made the GraphQL polling log engine a functor so that both the cloud and the local integrations can utilize it. In addition to making it a functor, I added a parameter to specify an interval to issue GraphQL requests to start collecting events, which is vitally important for the local version.

It's important because Lucy will wait for all nodes to initialize as a test condition. These conditions are checked by issuing GraphQL requests to the nodes to start collecting their events and make them available in a query. There is a race condition where a node can initialize before it's been set to start collecting it's logs, which means the node initialization event will be missed and the tests fail. To fix this, we make GraphQL requests much more frequent on the local version to make nodes immediately start collecting logs after the GraphQL endpoint is available.

I thought about making the GraphQL availability it's own structured event which we listen too, so that we can avoid this interval parameter and just issue a request to collect events right when GraphQL is available. I decided not to do that since we would need to land that change into different branches so that the daemon emits that event, and figured that this is an easier way to unblock this issue.

The local version issues requests every `0.25` seconds to nodes and the cloud version keeps the same `10` seconds.